src/Compilers/CSharp/Test/Semantic/Semantics/DelegateTypeTests.cs

```diff
@@ -0,0 +1,708 @@
+                Diagnostic(ErrorCode.ERR_AnonMethToNonDel, "() => { }").WithArguments("lambda expression", "System.ICloneable").WithLocation(6, 31),
+                // (8,13): error CS8915: The delegate type could not be inferred.
+                //         d = x => x;
+                Diagnostic(ErrorCode.ERR_CannotInferDelegateType, "x => x").WithLocation(8, 13));
+        }
+
+        public static IEnumerable<object?[]> GetMethodGroupData()
+        {
+            yield return getData("static int F() => 0;", "F", "System.Func<System.Int32>", "Func`1");
+            yield return getData("static ref int F() => throw null;", "F", null, null);
+            yield return getData("static ref readonly int F() => throw null;", "F", null, null);
+            yield return getData("static void F() { }", "F", "System.Action", "Action");
+            yield return getData("static void F(int x, int y) { }", "F", "System.Action<System.Int32, System.Int32>", "Action`2");
+            yield return getData("static void F(out int x, int y) { x = 0; }", "F", null, null);
+            yield return getData("static void F(int x, ref int y) { }", "F", null, null);
+            yield return getData("static void F(int x, in int y) { }", "F", null, null);
+            yield return getData("static void F(int _1, object _2, int _3, object _4, int _5, object _6, int _7, object _8, int _9, object _10, int _11, object _12, int _13, object _14, int _15, object _16) { }", "F", "System.Action<System.Int32, System.Object, System.Int32, System.Object, System.Int32, System.Object, System.Int32, System.Object, System.Int32, System.Object, System.Int32, System.Object, System.Int32, System.Object, System.Int32, System.Object>", "Action`16");
+            yield return getData("static void F(int _1, object _2, int _3, object _4, int _5, object _6, int _7, object _8, int _9, object _10, int _11, object _12, int _13, object _14, int _15, object _16, int _17) { }", "F", null, null);
+            yield return getData("static object F(int _1, object _2, int _3, object _4, int _5, object _6, int _7, object _8, int _9, object _10, int _11, object _12, int _13, object _14, int _15, object _16) => null;", "F", "System.Func<System.Int32, System.Object, System.Int32, System.Object, System.Int32, System.Object, System.Int32, System.Object, System.Int32, System.Object, System.Int32, System.Object, System.Int32, System.Object, System.Int32, System.Object, System.Object>", "Func`17");
+            yield return getData("static object F(int _1, object _2, int _3, object _4, int _5, object _6, int _7, object _8, int _9, object _10, int _11, object _12, int _13, object _14, int _15, object _16, int _17) => null;", "F", null, null);
+            yield return getData("T F<T>() => default;", "(new Program()).F<int>", "System.Func<System.Int32>", "Func`1");
+
+            static object?[] getData(string methodDeclaration, string methodGroupExpression, string? expectedType, string? expectedName) =>
```

Comment:
It looks like `methodGroupExpression` is always "F". Consider removing that parameter and hard-coding in tests

Reply:
The method declaration and method group are related, and at the moment, the test and the test data are mostly independent.